ml-algorithms/src/main/java/org/opensearch/ml/engine/processor/ProcessorChain.java

```
@@ -0,0 +1,556 @@
+            }
+
+            // Handle numeric conditions
+            if (value instanceof Number || canParseAsNumber(strValue)) {
+                double numValue;
+                if (value instanceof Number) {
+                    numValue = ((Number) value).doubleValue();
+                } else {
+                    try {
+                        numValue = Double.parseDouble(strValue);
+                    } catch (NumberFormatException e) {
+                        return false;
+                    }
+                }
+
+                // Check numeric conditions
+                if (condition.startsWith(">") && !condition.startsWith(">=")) {
+                    double threshold = Double.parseDouble(condition.substring(1));
+                    return numValue > threshold;
+                } else if (condition.startsWith("<") && !condition.startsWith("<=")) {
+                    double threshold = Double.parseDouble(condition.substring(1));
+                    return numValue < threshold;
+                } else if (condition.startsWith(">=")) {
+                    double threshold = Double.parseDouble(condition.substring(2));
+                    return numValue >= threshold;
+                } else if (condition.startsWith("<=")) {
+                    double threshold = Double.parseDouble(condition.substring(2));
+                    return numValue <= threshold;
+                } else if (condition.startsWith("==")) {
+                    double threshold = Double.parseDouble(condition.substring(2));
+                    return Math.abs(numValue - threshold) < 1e-10;
+                }
+            }
+
+            // Handle regex matching
+            if (condition.startsWith("regex:")) {
+                String regex = condition.substring(6);
+                try {
+                    return Pattern.matches(regex, strValue);
+                } catch (Exception e) {
+                    log.warn("Invalid regex in condition: {}", regex);
```

Should we throw an error here instead of failing silently?